packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -91,13 +91,13 @@ export const prepareStatusManager = (
   /**
    * Transactions seen *ever* by the contract.
    *
-   * Note that like all durable stores, this SetStore is stored in IAVL. It
-   * grows without bound (though the amount of growth per incoming message to
-   * the contract is bounded). At some point in the future we may want to prune.
-   * @type {SetStore<EvmHash>}
+   * Note that like all durable stores, this MapStore is kept in IAVL. It stores
+   * the `blockNumber` so that later we can prune old transactions.
+   * @type {MapStore<EvmHash, NatValue>}
    */
-  const seenTxs = zone.setStore('SeenTxs', {
+  const seenTxs = zone.mapStore('SeenTxs', {
     keyShape: M.string(),
+    valueShape: M.nat(),
```

Comment:
I wonder about schema evolution of the value shape...
I suppose we could change it at the cost of copying the map.
And changing it is pretty unlikely in any case; block height seems sufficient.

Reply:
oh. MH asked whether we're comparing block heights across chains. Yes, we would be. So maybe block height isn't sufficient.

I'm still OK merging this change; it might not be a complete solution, but it looks like forward progress.

Reply:
blockTimestamp should work.